master/internal/db/postgres_experiments.go

```diff
@@ -277,6 +277,47 @@ ORDER BY s.end_time;`, &rows, metricName, experimentID, minBatches, maxBatches,
 	return trials, endTime, nil
 }
 
+// GenericTrialsSnapshot returns metrics across each trial in an experiment at a
+// specific point of progress, for metric groups other than training and validation.
+func (db *PgDB) GenericTrialsSnapshot(experimentID int, minBatches int, maxBatches int,
```

Comment:
question: what are the differences between Training, Validation, and this generic version? can this replace the other two?

Also can we use the dedicated type for metric groups throughout?

Reply:
I'll use `model.MetricGroup` here too
My understanding is Training reads from `steps.metrics->'avg_metrics'->>'MetricName'`
Validation from `validations.metrics->'validation_metrics'->>'MetricName'`

And the generic has to read from the `metrics` table to filter on `metric_group`; otherwise it is more similar to Training with `metrics.metrics->'avg_metrics'->>'MetricName'`

Reply:
this helper should help with handling the difference between where you'd find the metrics 

https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/pkg/model/experiment.go#L525-L531

we should be able to have all of these read from the `metrics` tables with help from maybe
https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/internal/db/postgres_trial_metrics.go#L97-L110  and maybe similar to how [GetMetrics](https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/internal/db/postgres_trial_metrics.go#L292) works

otherwise I think everything looks good.

Reply:
👍🏼 now that there's only one version of these drop the Generic from the name?

Reply:
thanks for consolidating these.